pkg/api/vm.go

```diff
@@ -99,7 +99,7 @@ func (v *VMDef) adjustDiskBootIdx(qti *qcli.QemuTypeIndex) ([]string, error) {
 	// for any disks without a BootIndex, allocate one
 	for n := range v.Disks {
 		disk := v.Disks[n]
-		if disk.BootIndex == "" {
+		if disk.BootIndex == "" && disk.BootIndex != "off" {
```

Comment:
Hm, same would be true here.  It *feels* like you intend the first disk.BootIndex to be a newly-allocated bootindex, and the second to be user-provided...

Reply:
So, the concern here is if you declare a disk or a nic, and *dont* set a bootindex, then we calculate a bootindex for you; 

which I guess is OK, we're now explicitly telling machined to not allocate a bootindex for _this_ device; so if you have 4 nics, you'll have to turn bootindex off for each one.  That' seems fine.  

I'll refactor this.